recipes/yaml/CMakeLists.txt.patch

```diff
@@ -0,0 +1,25 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index e84c28c..eab70b2 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -3,6 +3,13 @@
+ cmake_minimum_required (VERSION 2.8)
+ project (yaml C)
+ 
++# Allow the developer to select if Dynamic or Static libraries are built
++OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
++SET (LIB_TYPE STATIC)
++IF (BUILD_SHARED_LIBS)
++    SET (LIB_TYPE SHARED)
++ENDIF (BUILD_SHARED_LIBS)
++
+ set (YAML_VERSION_MAJOR 0)
+ set (YAML_VERSION_MINOR 1)
+ set (YAML_VERSION_PATCH 6)
+@@ -12,5 +19,4 @@ file (GLOB SRC src/*.c)
+ 
+ include_directories (include win32)
+ add_definitions (-DHAVE_CONFIG_H -DYAML_DECLARE_STATIC)
+-add_library (yaml STATIC ${SRC})
+-
++add_library (yaml ${LIB_TYPE} ${SRC})
```

Comment:
You have to make sure the import lib for the DLL doesn't get overwritten by your static lib so probably safest to have different names. I was also unable to work out how to do one build for static/shared with cmake for jpeg. This might be because on Linux (and some other OS's) static and shared builds need different compile flags.